src/asm/rpn.c

```diff
@@ -50,7 +51,7 @@ static uint8_t *reserveSpace(struct Expression *expr, uint32_t size)
 			 * To avoid generating humongous object files, cap the
 			 * size of RPN expressions
 			 */
-			fatalerror("RPN expression cannot grow larger than %d bytes",
+			fatalerror("RPN expression cannot grow larger than %lu bytes",
 				   MAXRPNLEN);
```

Comment:
The cast to `unsigned long` should be here instead of adding the types to the constant. That way you avoid unnecessary promotion elsewhere.

Reply:
To quote directly:
> JL2210 Yesterday at 2:15 PM
> ISSOtm Which would you prefer: an integer constant macro with an `LU` appended to the number to make it an `unsigned long`, and then print with `%lu`, or an explicit cast to `unsigned long` and still print with `lu`?
> ISSOtm Yesterday at 2:19 PM
> The former
> More compact, imo still carries the point across
> Why'd you ask? I don't see anywhere this'd be useful in the codebase offhand
> JL2210 Yesterday at 2:22 PM
> printing MAXRPN, too big to fit in 16-bit `unsigned`

Reply:
CC: @ISSOtm

This is a bad idea because it splits two things that are clearly supposed to be correlated across files for no good reason. Saving ten characters doesn't really justify it.

Reply:
I imagine that ISSOtm probably didn't know that they were split across files. I'll change it.

Reply:
No, I didn't.

Reply:
Fixed. Is this good?

Reply:
Seems OK to me.